tests/templates/test_subroutines/test_select.py

```diff
@@ -0,0 +1,277 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for the UCCSD template.
+"""
+# pylint: disable=protected-access,too-many-arguments,import-outside-toplevel, no-self-use
+import pytest
+import numpy as np
+from pennylane import numpy as pnp
+import pennylane as qml
+
+
+class TestDecomposition:
+    """Tests that the template defines the correct decomposition."""
+
+    @pytest.mark.parametrize(
+        ("ops", "control_wires", "expected_gates"),
+        [
+            (
+                [qml.PauliX(wires=0), qml.PauliY(wires=0)],
+                [1],
+                [
+                    qml.ctrl(qml.PauliX(wires=0), control=1, control_values=0),
+                    qml.ctrl(qml.PauliY(wires=0), control=1),
+                ],
+            ),
+        ],
+    )
+    def test_operation_result(self, ops, control_wires, expected_gates):
+        """Test the correctness of the SELECT template output."""
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit1():
+            qml.SELECT(ops, control_wires)
+            return qml.state()
+
+        @qml.qnode(dev)
+        def circuit2():
+            for op in expected_gates:
+                qml.apply(op)
+            return qml.state()
+
+        assert np.allclose(circuit1(), circuit2())
+
+    @pytest.mark.parametrize(
+        ("ops", "control_wires", "expected_gates"),
+        [
+            (
+                [qml.PauliX(wires=0), qml.PauliY(wires=0)],
```

Comment:
Could we add a test that uses `Identity` for skipping a bit-string and another with custom wire labels?

Reply:
Added